src/storage/ArrayStorage.java

```diff
@@ -6,72 +6,103 @@
+public abstract class ArrayStorage implements Storage {
+    protected static final int NOT_EXISTING_INDEX = -1;
+    protected static final int MAX_STORAGE_SIZE = 100000;
 
+    protected final Logger log = Logger.getLogger(getClass().getSimpleName());
+
+    protected Resume[] storage;
+
+
+    protected ArrayStorage() {
+        this.storage = new Resume[MAX_STORAGE_SIZE];
+    }
+
+    protected int size;
+
+    @Override
     public void clear() {
         Arrays.fill(storage, 0, size, null);
         size = 0;
     }
 
+    @Override
     public void save(Resume r) {
         Objects.requireNonNull(r);
         if (size == MAX_STORAGE_SIZE) {
-            log.warning("Storage max limit is reached. Please, clear storage to add new elements.");
+            logStorageMaxLimit();
             return;
         }
         if (isPresent(r.getUuid())) {
-            log.warning(String.format("Resume(uuid=%s) is already present in storage. It may be updated only.", r.getUuid()));
+            logSavePresentError(r.getUuid());
         } else {
-            storage[size] = r;
+            int indexToStoreAt = getResumeIndex(r.getUuid());
+            store(r, indexToStoreAt);
             size++;
         }
     }
 
+    @Override
     public void update(Resume r) {
         Objects.requireNonNull(r);
-        int index = getResumeIndexById(r.getUuid());
+        int index = getResumeIndex(r.getUuid());
         if (isValidIndex(index)) {
             storage[index] = r;
         } else {
-            log.warning(String.format("Resume(uuid=%s) is not present in storage. Please, add it.", r.getUuid()));
+            logNotFound(r.getUuid());
         }
     }
 
+    @Override
     public void delete(String uuid) {
         if (isEmpty(uuid)) {
-            log.warning(String.format("Delete impossible. Incorrect incoming parameter(uuid=%s)", uuid));
+            logNotFound(uuid);
             return;
         }
-        int index = getResumeIndexById(uuid);
+        int index = getResumeIndex(uuid);
         if (isValidIndex(index)) {
-            storage[index] = storage[size - 1];
-            storage[size - 1] = null;
+            erase(index);
             size--;
         } else {
-            log.warning(String.format("Delete impossible. Resume(uuid=%s) is not present in storage", uuid));
+            logNotFound(uuid);
         }
 
     }
 
+    @Override
     public Resume get(String uuid) {
         if (isEmpty(uuid)) {
-            log.warning(String.format("Invalid input parameters, empty UUID", uuid));
+            logEmptyUUID();
             return null;
         }
-        int index = getResumeIndexById(uuid);
+        int index = getResumeIndex(uuid);
         if (isValidIndex(index)) {
             return storage[index];
         }
-        log.warning(String.format("Resume(uuid=%s) is not present in storage. Return null.", uuid));
+        logNotFound(uuid);
         return null;
     }
 
+    protected void logEmptyUUID() {
```

Comment:
This method can be deleted. Use `logNotFound()`

Reply:
Done